include/deal.II/lac/trilinos_sparse_matrix.h

```diff
@@ -2036,10 +2046,412 @@ namespace TrilinosWrappers
+         * Returns an IndexSet that defines the partitioning of the domain space
+         * of this matrix, i.e., the partitioning of the vectors this matrix has
+         * to be multiplied with / operate on.
+         */
+        IndexSet
+        locally_owned_domain_indices () const;
+
+        /**
+         * Returns an IndexSet that defines the partitioning of the range space
+         * of this matrix, i.e., the partitioning of the vectors that result
+         * from matrix-vector products.
+         */
+        IndexSet
+        locally_owned_range_indices () const;
+
+        /**
+         * Return the MPI communicator object in use with this Payload.
+         */
+        MPI_Comm
+        get_mpi_communicator () const;
+
+        /**
+         * Sets an internal flag so that all operations performed by the matrix,
+         * i.e., multiplications, are done in transposed order.
+         * @note This does not reshape the matrix to transposed form directly,
+         * so care should be taken when using this flag.
+         */
+        void
+        transpose ();
+
+        /**
+         * The standard matrix-vector operation to be performed by the payload
+         * when Apply is called.
+         *
+         * @note This is not called by a LinearOperator, but rather by Trilinos
+         * functions that expect this to us mimic the action of the
+         * LinearOperator.
+         */
+        std::function<void(VectorType &, const VectorType &)> vmult;
+
+        /**
+         * The standard transpose matrix-vector operation to be performed by
+         * the payload when Apply is called.
+         *
+         * @note This is not called by a LinearOperator, but rather by Trilinos
+         * functions that expect this to us mimic the action of the
+         * LinearOperator.
+         */
+        std::function<void(VectorType &, const VectorType &)> Tvmult;
+
+        /**
+         * The inverse matrix-vector operation to be performed by the payload
+         * when ApplyInverse is called.
+         *
+         * @note This is not called by a LinearOperator, but rather by Trilinos
+         * functions that expect this to us mimic the action of the
+         * InverseOperator.
+         */
+        std::function<void(VectorType &, const VectorType &)> inv_vmult;
+
+        /**
+         * The inverse transpose matrix-vector operation to be performed by
+         * the payload when ApplyInverse is called.
+         *
+         * @note This is not called by a LinearOperator, but rather by Trilinos
+         * functions that expect this to us mimic the action of the
+         * InverseOperator.
+         */
+        std::function<void(VectorType &, const VectorType &)> inv_Tvmult;
+
+//@}
+
+        /**
+        * @name Core Epetra_Operator functionality
+        */
+//@{
+
+        /**
+         * Return the status of the transpose flag for this operator
+         */
+        virtual bool
+        UseTranspose () const;
```

Comment:
this and the functions below do not match deal.II function naming conventions. Is this because you overload functions from `Epetra_Operator`? If so, maybe just add "Overloads the same function from the Trilinos class Epetra_Operator." here and below.